### ISSUE TITLE
PIP installation missing numpy requirement

### ISSUE THREAD

**Issue by pedro-ricardo** [dmuellner/fastcluster#23]
Hello there,
FastCluster appears to use a `numpy>1.20`. But this dependency is not set in PIP therefore if some other package needs a `numpy<1.19` ,for example, it will install but give the following error upon usage:
```bash
RuntimeError: module compiled against API version 0xe but this version of numpy is 0xd
(...)
    from fastcluster import linkage
    from _fastcluster import linkage_wrap, linkage_vector_wrap
ImportError: numpy.core.multiarray failed to import
```

I believe the correct behavior in this case would be not installing due to dependency conflict and PIP would detect and warn about it.

**Comment by pedro-ricardo** [comment]
To reproduce this behavior you can set a requirements in a virtual environment with `numpy<1.20` and `fastcluster`, that just import it to see.

The original error I got when using `fastcluster` and  `tensorflow` in the same requirements. As `tensorflow` asks for a `numpy` between `1.16` and `1.19` it fails on runtime just like mentioned.

**Comment by dmuellner** [comment]
Can you check whether fastcluster 1.2.4 solves your problem?

**Comment by dmuellner** [comment]
I assume that this problem was fixed in fastcluster 1.2.4. Please re-open this issue if this is not the case.
